Framework/DataHandling/test/StartAndEndTimeFromNexusFileExtractorTest.h

```diff
@@ -0,0 +1,118 @@
+  static StartAndEndTimeFromNexusFileExtractorTest *createSuite() {
+    return new StartAndEndTimeFromNexusFileExtractorTest();
+  }
+  static void destroySuite(StartAndEndTimeFromNexusFileExtractorTest *suite) {
+    delete suite;
+  }
+
+  void test_that_throws_for_non_sense_file() {
+    // Arrrange
+    StartAndEndTimeFromNexusFileExtractor extractor;
+    std::string filename = "file_doesnt_exist.nxs";
+
+    // Act + Assert
+    TSM_ASSERT_THROWS_ANYTHING(
+        "Should throw something when opening file which does not exist",
+        extractor.extractEndTime(filename));
+  }
+
+  void test_that_times_can_be_extracted_from_muon_file() {
+    // Arrange
+    StartAndEndTimeFromNexusFileExtractor extractor;
+    std::string filename = "emu00006473.nxs";
+    auto fullFilePath =
+        Mantid::API::FileFinder::Instance().getFullPath(filename);
+
+    // Act
+    auto startTime = extractor.extractStartTime(fullFilePath);
+    auto endTime = extractor.extractEndTime(fullFilePath);
+
+    // Assert
+    Mantid::Kernel::DateAndTime expectedStartTimeString("2006-11-21T07:04:30");
+    Mantid::Kernel::DateAndTime expectedEndTimeString("2006-11-21T09:29:28");
+    TSM_ASSERT("Should have the same start time",
+               startTime == expectedStartTimeString);
+    TSM_ASSERT("Should have the same end time",
+               endTime == expectedEndTimeString);
+  }
+
+  void test_that_times_can_be_extracted_from_isis_file() {
+    // Arrange
+    StartAndEndTimeFromNexusFileExtractor extractor;
+    std::string filename = "POLREF00014966.nxs";
+    auto fullFilePath =
+        Mantid::API::FileFinder::Instance().getFullPath(filename);
+
+    // Act
+    auto startTime = extractor.extractStartTime(fullFilePath);
+    auto endTime = extractor.extractEndTime(fullFilePath);
+
+    // Assert
+    Mantid::Kernel::DateAndTime expectedStartTimeString("2015-10-13T05:34:32");
+    Mantid::Kernel::DateAndTime expectedEndTimeString("2015-10-13T11:30:28");
+    TSM_ASSERT("Should have the same start time",
+               startTime == expectedStartTimeString);
+    TSM_ASSERT("Should have the same end time",
+               endTime == expectedEndTimeString);
+  }
+
+  void test_that_times_can_be_extracted_from_processed_file() {
+    // Arrange
+    StartAndEndTimeFromNexusFileExtractor extractor;
+    std::string filename = "LOQ48127.nxs";
+    auto fullFilePath =
+        Mantid::API::FileFinder::Instance().getFullPath(filename);
+
+    // Act
+    auto startTime = extractor.extractStartTime(fullFilePath);
+    auto endTime = extractor.extractEndTime(fullFilePath);
+
+    // Assert
+    Mantid::Kernel::DateAndTime expectedStartTimeString("2008-12-18T17:58:38");
+    Mantid::Kernel::DateAndTime expectedEndTimeString("2008-12-18T18:06:20");
+    TSM_ASSERT("Should have the same start time",
+               startTime == expectedStartTimeString);
+    TSM_ASSERT("Should have the same end time",
+               endTime == expectedEndTimeString);
```

Comment:
The same test is repeated, refactor to a method (with filename, expected start and expected end arguments)?